### PR TITLE
feat: computed members added

### PR DESCRIPTION
**Pull request by waxb**
- chore: changelog updated
- chore!: update module from template and tests
- feat: computed members added

### CODE REVIEW

**Review by zied-elouaer — DISMISSED** [review]
LGTM

File: iam.tf [hunk 1]
```diff
@@ -26,7 +26,9 @@ module "iam" {
 
   role = each.value.role
 
-  members       = try(each.value.members, [])
+  members       = try(local.iam[each.value].members, [])
   condition     = try(each.value.condition, null)
   authoritative = try(each.value.authoritative, true)
```
Comment (mariux) [inline iam.tf:31]: this looks odd as we have not converted to using index here.. please fix the map to point to the original index and use index everywhere..

File: variables.tf [hunk 2]
```diff
@@ -68,6 +68,17 @@ variable "auto_create_network" {
...
+  type        = map(string)
+  description = "(Optional) A map of members to replace in 'members' to handle terraform computed values. Will be ignored when policy bindings are used."
+  default     = {}
+
+  validation {
+    condition     = alltrue([for k, v in var.computed_members_map : can(regex("^(allUsers|allAuthenticatedUsers|(user|serviceAccount|group|domain):)", v))])
+    error_message = "The value must be a non-empty string being a valid principal type identified with `allUsers`, `allAuthenticatedUsers` or prefixed with `user:`, `serviceAccount:`, `group:`, or `domain:`."
+  }
+}
+
```
Comment (mariux) [inline variables.tf:81]: validation is wrong. please check with iam module validation.

**Review by mariux — CHANGES_REQUESTED** [review]

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**